test/inductor/test_triton_kernels.py

```diff
@@ -1630,6 +1636,158 @@ def f(x, y, z):
         self.assertEqual(out2, x + y + 1)
         self.assertEqual(out3, z**2)
 
+    @requires_gpu
+    @unittest.skipIf(not has_triton_tma(), "requires Triton TMA support")
+    def test_tma_capture_and_functionalize(self):
+        def f(a, b):
+            BLOCK_SIZE = 256
+            out = torch.zeros_like(a)
+            n_elements = out.numel()
+
+            desc_a, desc_b, desc_out = (
+                triton.tools.experimental_descriptor.create_1d_tma_descriptor(
+                    t.data_ptr(),
+                    n_elements,
+                    BLOCK_SIZE,
+                    t.element_size(),
+                )
+                for t in (a, b, out)
+            )
+
+            grid = lambda meta: (triton.cdiv(n_elements, meta["BLOCK_SIZE"]),)
+            add_kernel_with_tma_1d[grid](
+                desc_a,
+                desc_b,
+                desc_out,
+                BLOCK_SIZE=BLOCK_SIZE,
+            )
+
+            return out
+
+        a = torch.randn(301, device=GPU_TYPE)
+        b = torch.randn(301, device=GPU_TYPE)
+
+        torch._logging.set_logs(aot=logging.INFO)
+        with self.assertLogs(
+            logger="torch._functorch._aot_autograd.dispatch_and_compile_graph.__aot_graphs"
+        ) as log:
+            compiled_out = torch.compile(
+                f,
+                fullgraph=True,
+                backend="aot_eager",
+                dynamic=False,
+            )(a, b)
+        # reset logging state
+        torch._logging.set_logs()
+
+        output = "\n".join(record.getMessage() for record in log.records)
+        output = re.sub(r"\033\[[0-9;]*m", "", output)  # remove coloring
+
+        # structured capturing of the TMA descriptor metadata
+        self.assertIn(
+            "tma_descriptor_metadata = {'in_desc_ptr0': ([301], [256], 4), 'in_desc_ptr1': ([301], [256], 4), 'out_desc_ptr': ([301], [256], 4)}",
+            output,
+        )
+        # functionalization: the output should be marked as mutated
+        self.assertIn("tensors_to_clone = ['out_desc_ptr']", output)
```

Comment:
is it possible to just do an assertExpectedInline? This makes it clearer what's going on overall with the design.

For example: "out_desc_ptr" is a key in both the tma_descriptor_metadata and the kwargs dicts passed to triton_kernel_wrap. I learned this the other day when you sent me some graphs; putting the graphs here in code would help other people figure out what's going on